lib/AST/Type.cpp

```diff
@@ -905,6 +905,25 @@ unsigned int TypeBase::getOptionalityDepth() {
 }
 
 Type TypeBase::stripConcurrency(bool recurse, bool dropGlobalActor) {
+  if (auto *arrayTy = dyn_cast<ArraySliceType>(this)) {
```

Comment:
This change is not technically tied for this PR since matching happens on canonical types, you can land this separately.